sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -930,6 +930,16 @@ class CodegenContext {
       // inline execution if only one block
       blocks.head
     } else {
+      if (Utils.isTesting) {
+        // Passing global variables to the split method is dangerous, as any mutating to it is
+        // ignored and may lead to unexpected behavior.
+        val mutableStateNames = mutableStates.map(_._2).toSet
```

Comment:
after finally merging SPARK-18016, this should be the union of `arrayCompactedMutableStates.flatMap(_.arrayNames)` and `inlinedMutableStates.map(_._2)`, I think